### PR TITLE
introduce hudson.Util#fixNull(value, defaultValue)

### PR DESCRIPTION
**Pull request by ndeloof**
* internal: introduce fixNull(value, defaultValue)

- [ ] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

File: core/src/main/java/hudson/Util.java [hunk 1]
```diff
@@ -1195,8 +1195,14 @@ public static void copyFile(@Nonnull File src, @Nonnull File dst) throws BuildEx
...
     @Nonnull
     public static String fixNull(@CheckForNull String s) {
-        if(s==null)     return "";
-        else            return s;
+        return fixNull(s, "");
+    }
+
+    /**
+     * Convert null to defaultValue.
+     */
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/Util.java]: `@since TODO`

**Review by oleg-nenashev — APPROVED** [review]
Works for me though it may be already supported by existing libraries bundled into Jenkins

**Review by batmat — COMMENTED** [review]
Thanks for adding these.
I'm aware it's trivial code, but still, adding associated tests looks important to me: it's core code that could will likely end up being used dozens or hundreds of times. For usual reasons, having tests for this is important if not critical/required IMO.

File: core/src/main/java/hudson/Util.java [hunk 2]
```diff
@@ -1195,8 +1195,14 @@ public static void copyFile(@Nonnull File src, @Nonnull File dst) throws BuildEx
...
+    public static <T> T fixNull(@CheckForNull T s, T defaultValue) {
```
Comment (batmat) [inline core/src/main/java/hudson/Util.java]: @NonNull on second param and return type?

**Comment by batmat** [comment]
`org.apache.commons.lang.ObjectUtils#defaultIfNull` or `org.apache.commons.lang3.ObjectUtils#defaultIfNull`

**Review by Wadeck — APPROVED** [review]
Note for the code around: it's weird that we are returning unmodifiable collection without noting that on the JavaDoc

**Comment by oleg-nenashev** [comment]
Addressed Javadoc/annotation comments for the new method.
@Wadeck could you please create a follow-up newbie-friendly ticket for collections?

**Comment by Wadeck** [comment]
@oleg-nenashev => https://issues.jenkins-ci.org/browse/JENKINS-53847

**Pull request merged**